examples/all-clusters-app/all-clusters-common/all-clusters-app.zap

```diff
@@ -5633,6 +5593,196 @@
             }
           ]
         },
+        {
+          "name": "Network Provisioning",
+          "code": 43690,
+          "mfgCode": null,
+          "define": "NETWORK_PROVISIONING_CLUSTER",
+          "side": "client",
```

Comment:
Please disable the `client` side here for the `all-clusters-app`. It has been enabled in `chip-tool` already.

Reply:
:-(
https://github.com/project-chip/connectedhomeip/blob/3e6f2d43b0609f4aa7e3ad7a361781483016f4a5/scripts/tools/zap_generate_chip.sh#L21

Reply:
Yes. As I said in your IM patch I'm trying to get rid of `chip-templates.json` in order to let applications uses their own `.zap`file.

So that means the python device controller should have its own `.zap` file and can have custom scripts to generate the `CHIPDeviceController-*` bits.

It also means that I need to move `CHIPClusters*-zap` files to be application specific. So applications will just generate the right set of methods based on what is in their `.zap` file.

I also project to move some of the content of `src/app/encoder.cpp` and `src/app/chip-zcl-zpro-codec-api.h` inside `src/controller/CHIPClusters*` (which would be `gen/CHIPClusters*`. files). And mark those files as deprecated since they can't be removed until all the consumers stop using them. 
For this part I somehow hope to find a way to switch to the new IM instead of doing that since it just seems temporary work...

Finally we would need to find a way to make `CHIPClusters*` not glued to the controller CHIPDevice, maybe to a base class that `src/controller/CHIPDevice` inherits from,  so it can be used from the accessory side as well.

Reply:
I copied `all-clusters-app.zap` to `src/controller/controller-clusters.zap`, which makes sense for me.